enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/InternalFlatFileRealm.java

```diff
@@ -536,7 +563,10 @@ public void suspendUser( String username ) throws IOException, InvalidArgumentsE
             User suspendedUser = user.augment().withFlag( IS_SUSPENDED ).build();
             try
             {
-                userRepository.update( user, suspendedUser );
+                synchronized ( this )
```

Comment:
The comment block a couple of lines up says that this method is not synchronized. Maybe that should be updated?